### PR TITLE
pscanrules: More example alerts

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- CHANGELOGS > Added change notes.
- Scan rules > Added example alert functionality.
- UnitTests > Added tests for example alerts.

## Related Issues
- zaproxy/zaproxy#6119
- zaproxy/zaproxy#7100
- zaproxy/zaproxy#8262

## Checklist
- [NA] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRule.java [hunk 1]
```diff
@@ -293,6 +284,31 @@ public int getWascId() {
...
+    public List<Alert> getExampleAlerts() {
+        return List.of(
+                buildAlert(
+                                Alert.RISK_MEDIUM,
+                                Constant.messages.getString("pscanrules.noanticsrftokens.desc"),
+                                getExtraInfo(
+                                        "[token, csrfToken, csrf-token]", "[Form 1: \"name\" ]"),
+                                "<form name=\"someName\" data-no-csrf>")
+                        .build());
+    }
```
Comment (kingthorin) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRule.java:310]: Should I add handling and alert refs for the annotation case?

Comment (thc202) [reply addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRule.java:310]: IMO it's the same alert, but I don't mind either.

File: addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRuleUnitTest.java [hunk 2]
```diff
@@ -416,6 +430,8 @@ void shouldRaiseMediumAlertWhenFormAttributeAndRuleConfigMismatch() {
         // Then
         assertEquals(1, alertsRaised.size());
         assertEquals(Alert.RISK_MEDIUM, alertsRaised.get(0).getRisk());
+        System.out.println(alertsRaised.get(0).getOtherInfo());
+        System.out.println(alertsRaised.get(0).getEvidence());
```
Comment (thc202) [inline addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRuleUnitTest.java:434]: Remove.

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRule.java (hunk 1 above)
Comment (kingthorin) [reply addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRule.java:310]: Let's leave it for now, if a user brings it up we can adjust.

File: addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/UserControlledJavascriptEventScanRuleUnitTest.java [hunk 3]
```diff
@@ -162,7 +162,6 @@ void shouldRaiseAlertIfRequestParamValuesUsedInJsEvent() throws Exception {
         // Then
         assertThat(alertsRaised.size(), equalTo(1));
         assertThat(alertsRaised.get(0).getParam(), equalTo("name"));
-        System.out.println(alertsRaised.get(0).getOtherInfo());
```
Comment (kingthorin) [inline addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/UserControlledJavascriptEventScanRuleUnitTest.java:165]: Oops, musta previously snuck through.

**Comment by kingthorin** [comment]
Fixed

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**